scripts/bump_version.py

```diff
@@ -0,0 +1,28 @@
+# This script bumps up the version in setup.py for new releases.
+# Usage: python bump_version.py {new_version} (--setup_py_path <path-to setup.py>)
+
+import argparse
+import re
+
+parser = argparse.ArgumentParser(
```

Comment:
To conform a bit more with the conventions of this repo (and python best practices) I recommend that all the code from line 7 be wrapped inside of a function:
```python
def bump_version():
    parser = argparse.ArgumentParser(
        description="updates the version for a release",
        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
    )
    ... etc. ....
```
and then at the bottom call this function with:

```python
if __name__ == "__main__":
    bump_version()
```

See for example `generate_init.py` in the same directory.

Reply:
Thanks for the tips! I should follow python best practices.

Reply:
Please take a look at the updated script. I'm open to suggestions to make it better.

Reply:
oh wait hold on. I should run lint